frappe/public/js/frappe/form/controls/base_control.js

```diff
@@ -160,7 +160,13 @@ frappe.ui.form.Control = class BaseControl {
 	validate_and_set_in_model(value, e) {
 		var me = this;
 		let force_value_set = (this.doc && this.doc.__run_link_triggers);
-		let is_value_same = (this.get_model_value() === value);
+		let model_value = this.get_model_value();
+
+		if (this.df && this.df.fieldtype == 'Datetime') {
+			model_value = frappe.datetime.get_datetime_as_string(model_value);
+		}
```

Comment:
We should avoid fieldtype specific logic in base control.
you can move this to `datetime.js`